### PR TITLE
Cc

### PR DESCRIPTION
**Pull request by funman**

### CODE REVIEW

File: input/netmap/netmap.c [hunk 1]
```diff
@@ -623,28 +922,360 @@ static int catch_audio(struct uprobe *uprobe, struct upipe *upipe,
...
+    if (!anc_frame) {
+        netmap_ctx->anc_frame = anc_frame = new_coded_frame(output_stream->output_stream_id, 65536);
+        if (!anc_frame)
+            return UBASE_ERR_ALLOC;
+
+        anc_frame->len = 0;
+    }
+
+    bs_t s;
+    bs_init(&s, &anc_frame->data[anc_frame->len], 65536 - anc_frame->len);
```
Comment (kierank) [inline input/netmap/netmap.c:1011]: Why 65536, do I miss something here?

File: input/netmap/netmap.c [hunk 2]
```diff
@@ -623,28 +922,360 @@ static int catch_audio(struct uprobe *uprobe, struct upipe *upipe,
...
+
+            bool c_not_y = false;
+
+            bs_t s;
+            bs_init(&s, &anc_frame->data[anc_frame->len], 65536 - anc_frame->len);
+
+            bs_write(&s, 6, 0);
+            bs_write(&s, 1, c_not_y);
+            bs_write(&s, 11, vanc_line_number(fmt, i));
+            bs_write(&s, 12, hsize - h_left + get_vanc_offset(netmap_ctx));
```
Comment (kierank) [inline input/netmap/netmap.c]: This offset is relative to SAV, get_vanc_offset seems to be relative to EAV

File: input/netmap/netmap.c [hunk 3]
```diff
@@ -891,6 +1552,18 @@ static int open_netmap( netmap_ctx_t *netmap_ctx )
         upipe_release(vbi);
     }
 
+    /* vbi filter */
+    struct upipe_mgr *upipe_filter_vbi_mgr = upipe_vbi_mgr_alloc();
+    struct upipe *probe_vbi = upipe_vbi_alloc_output(vbi,
+            upipe_filter_vbi_mgr,
+            uprobe_pfx_alloc(uprobe_use(uprobe_dejitter), loglevel, "vbi filter"),
+            uprobe_pfx_alloc(uprobe_obe_alloc(uprobe_use(uprobe_dejitter), catch_ttx, netmap_ctx),
+               UPROBE_LOG_DEBUG, "op47"),
```
Comment (kierank) [inline input/netmap/netmap.c]: op47 is not the same as vbi

File: input/netmap/netmap.c (hunk 1 above)
Comment (kierank) [reply input/netmap/netmap.c:1011]: oh worst case PES packet size

Comment (funman) [reply input/netmap/netmap.c:1011]: yes

File: input/netmap/netmap.c (hunk 3 above)
Comment (funman) [reply input/netmap/netmap.c]: Seems to work though

File: input/netmap/netmap.c (hunk 2 above)
Comment (funman) [reply input/netmap/netmap.c]: Is it ?

Comment (funman) [reply input/netmap/netmap.c]: It doesn't make much but it seems so

Comment (funman) [reply input/netmap/netmap.c]: How would you write HANC on line 1, horizontal_offset = 1920+16/2 and line_number = 1125 on previous frame?

File: input/netmap/netmap.c (hunk 3 above)
Comment (funman) [reply input/netmap/netmap.c]: Do you want to change the debug "op47" into "ttx" ?

Comment (kierank) [reply input/netmap/netmap.c]: vbi_ttx

**Comment by kierank** [comment]
Why are there two functions to write HANC?

**Comment by funman** [comment]
One for HANC and one for VANC

**Comment by kierank** [comment]
Why does the VANC one contain HANC code?

**Comment by funman** [comment]
I guess last commit answers your question?

**Pull request merged**